src/api/teams.rs

```diff
@@ -37,3 +49,61 @@ pub async fn create(user: User, team: Json<NewTeam>, conn: DbConn) -> Result<Jso
     })
     .await
 }
+
+#[get("/teams/<team_slug>")]
+pub async fn team(team_slug: String, user: User, conn: DbConn) -> Result<Json<Team>, Status> {
+    conn.run(move |c| {
+        use crate::schema::team_users::dsl::{team_users, user_id};
+        use crate::schema::teams::dsl::{slug, teams};
+
+        let team = teams
+            .filter(slug.eq(team_slug).and(user_id.eq(user.id)))
+            .inner_join(team_users)
+            .first::<(Team, TeamUser)>(c)
+            .map_err(|e| {
+                if e == NotFound {
+                    Status::NotFound
+                } else {
+                    Status::InternalServerError
+                }
+            })
+            .map(|x| x.0)?;
+
+        Ok(Json(team))
+    })
+    .await
+}
+
+#[get("/teams/<team_slug>/users")]
+pub async fn users(team_slug: String, user: User, conn: DbConn) -> Result<Json<Vec<User>>, Status> {
+    conn.run(move |c| {
+        use crate::schema::team_users::dsl::{team_id, team_users, user_id};
+        use crate::schema::teams::dsl::{slug, teams};
+        use crate::schema::users::dsl::users;
+
+        // Fetch the team
+        let team = teams
+            .filter(slug.eq(&team_slug).and(user_id.eq(user.id)))
```

Comment:
it shouldn't be necessary here? just filter for equal to team slug

Reply:
Both are necessary so we can find a team that A) matches the slug and B) the current user is a member of